pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -299,3 +603,37 @@ func (r *ReplicatedUpgradeReconciler) duplicateSubclusterForReplicaGroupB(
 	baseSc.Annotations[vmeta.ChildSubclusterAnnotation] = newSc.Name
 	return newSc
 }
+
+// isReplicatedUpgradeSandboxCreated will check that the sandbox created for
+// replicated upgrade exists in the status.
+func (r *ReplicatedUpgradeReconciler) isReplicatedUpgradeSandboxCreated() (bool, error) {
+	scNames := r.getSubclustersForReplicaGroup(vmeta.ReplicaGroupBValue)
+	if len(scNames) == 0 {
+		return false, errors.New("cound not find any subclusters for replica group B")
+	}
+
+	foundSandbox := false
+	for i := range r.VDB.Status.Sandboxes {
+		sb := r.VDB.Status.Sandboxes[i]
+		if sb.Name != r.sandboxName {
+			continue
+		}
+		foundSandbox = true
+
+		// Sandbox was found. Do additional verification that the subclusters
+		// contained within it are correct.
+		if len(sb.Subclusters) != len(scNames) {
```

Comment:
Can you add what can cause this?

Reply:
This is just defensive programming. This should always be the case. But its good practice to check in case of some programming error.